<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Alessandro Romanel, Davide Dalfovo" />

<meta name="date" content="2023-01-10" />

<title>EthSEQ: Ethnicity Annotation from Whole-Exome and Targeted Sequencing Data</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">EthSEQ: Ethnicity Annotation from
Whole-Exome and Targeted Sequencing Data</h1>
<h4 class="author">Alessandro Romanel, Davide Dalfovo</h4>
<h4 class="date">2023-01-10</h4>



<p>Whole-exome sequencing (WES) and targeted sequencing (TS) are widely
utilized both in translational cancer genomics studies and in the
setting of precision medicine. Stratification of individuals’
ethnicity/ancestry is fundamental for the correct interpretation of
personal genomic variation impact. We implemented EthSEQ to provide
reliable and rapid ancestry annotation from whole -exome and targeted
sequencing data. EthSEQ can be integrated into any WES or TS based
processing pipeline and exploits multi-core capabilities.</p>
<p>EthSEQ requires genotype data at SNPs positions for a set of
individuals with known ancestry (the reference model) and either a list
of BAM files or genotype data (in VCF or GDS formats) of individuals
with unknown ancestry (the target model). EthSEQ annotates the ancestry
of each individual using an automated procedure and returns detailed
information about individual’s inferred ancestry, including aggregated
visual reports.</p>
<hr />
<div id="perform-ethnicity-analysis-with-individuals-genotype-data-from-vcf-file" class="section level2">
<h2>Perform ethnicity analysis with individuals genotype data from VCF
file</h2>
<p>Analysis of target individuals genotype data exploiting a reference
model built from 1,000 Genome Project genotype data. Genotype data for
10,000 exonic SNPs are provided in input to EthSEQ in VCF format while
the reference model is provided in GDS format and describes genotype
data for 1,000 Genome Project individuals for the same SNPs set.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(EthSEQ)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Run the analysis</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ethseq.Analysis</span>(</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">target.vcf =</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;Samples.HGDP.10000SNPs.vcf&quot;</span>,<span class="at">package=</span><span class="st">&quot;EthSEQ&quot;</span>),</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">model.gds =</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;Reference.Gencode.Exome.10000SNPs.gds&quot;</span>,<span class="at">package=</span><span class="st">&quot;EthSEQ&quot;</span>),</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">out.dir =</span> <span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;EthSEQ_Analysis/&quot;</span>),</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose=</span><span class="cn">TRUE</span>,</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">cores =</span><span class="dv">1</span>,</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">composite.model.call.rate =</span> <span class="dv">1</span>,</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">space =</span> <span class="st">&quot;3D&quot;</span>)</span></code></pre></div>
<pre><code>## [2023-01-10 17:19:30] Running EthSEQ
## [2023-01-10 17:19:30] Working directory: /tmp/Rtmpk6qR2o/EthSEQ_Analysis/
## [2023-01-10 17:19:30] Create /tmp/Rtmpk6qR2o/EthSEQ_Analysis/ folder
## [2023-01-10 17:19:30] Create target model from VCF
## [2023-01-10 17:19:30] Create aggregated model
## Merge SNP GDS files:
##     open &#39;/tmp/Rtmpk6qR2o/EthSEQ_Analysis//Target.gds&#39; ...
##         6 samples, 10000 SNPs
##     open &#39;/tmp/Rtmp1RDx5G/Rinst7a5b1f548fdc/EthSEQ/extdata/Reference.Gencode.Exome.10000SNPs.gds&#39; ...
##         619 samples, 10000 SNPs
## Concatenating samples (mapping to the first GDS file) ...
##     reference: 10000 SNPs (100.0%)
##     file 2: 586 allele flips, 0 ambiguous locus/loci
##         [no flip]: 9414
##         [flip]: 586
##     create &#39;/tmp/Rtmpk6qR2o/EthSEQ_Analysis//Aggregated.gds&#39;: 625 samples, 10000 SNPs
##     FileFormat = SNP_ARRAY
##     writing genotypes ...
##     transposing the genotype matrix ...
## Clean up the fragments of GDS file:
##     open the file &#39;/tmp/Rtmpk6qR2o/EthSEQ_Analysis//Aggregated.gds&#39; (1.8M)
##     # of fragments: 35
##     save to &#39;/tmp/Rtmpk6qR2o/EthSEQ_Analysis//Aggregated.gds.tmp&#39;
##     rename &#39;/tmp/Rtmpk6qR2o/EthSEQ_Analysis//Aggregated.gds.tmp&#39; (1.1M, reduced: 718.8K)
##     # of fragments: 15
## Done.
## [2023-01-10 17:19:31] Perform PCA on aggregated model
## Principal Component Analysis (PCA) on genotypes:
## Excluding 0 SNP on non-autosomes
## Excluding 543 SNPs (monomorphic: TRUE, MAF: NaN, missing rate: 0)
##     # of samples: 625
##     # of SNPs: 9,457
##     using 1 thread
##     # of principal components: 5
## PCA:    the sum of all selected genotypes (0,1,2) = 9939848
## CPU capabilities: Double-Precision SSE2
## Tue Jan 10 17:19:31 2023    (internal increment: 1616)
## 
[..................................................]  0%, ETC: ---        
[==================================================] 100%, completed, 1s
## Tue Jan 10 17:19:32 2023    Begin (eigenvalues and eigenvectors)
## Tue Jan 10 17:19:32 2023    Done.
## [2023-01-10 17:19:32] Infer ethnicities
## [2023-01-10 17:19:32] Print annotations
## [2023-01-10 17:19:32] Plot visual report</code></pre>
<pre><code>## [2023-01-10 17:19:32] Computation end</code></pre>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Load and display computed ethnicity annotations</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>ethseq.annotations <span class="ot">=</span> <span class="fu">read.delim</span>(<span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;EthSEQ_Analysis/Report.txt&quot;</span>),</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">as.is=</span><span class="cn">TRUE</span>,<span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ethseq.annotations)</span></code></pre></div>
<pre><code>##   sample.id pop    type            contribution
## 1   Sample1 AFR  INSIDE                        
## 2   Sample2 AFR  INSIDE                        
## 3   Sample3 EUR  INSIDE                        
## 4   Sample4 EUR CLOSEST   EUR(89.3%)|AFR(10.7%)
## 5   Sample5 SAS CLOSEST EUR(28.21%)|SAS(71.79%)
## 6   Sample6 EAS  INSIDE</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Delete analysis folder</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">unlink</span>(<span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;EthSEQ_Analysis/&quot;</span>),<span class="at">recursive=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<p>Current version of EthSEQ manages only VCF files with the following
format: - FORMAT column should contain “GT” - Only genotypes 0/0, 0/1,
1/1 and ./. are admitted - Only positions with single reference and
single alternative base are admitted - No duplicate IDs are admitted (so
no multiple variants with ID equal to “.”) - No duplicated sample names
are admitted - No duplicated positions are admitted</p>
</div>
<div id="perform-ethnicity-analysis-using-pre-computed-reference-model" class="section level2">
<h2>Perform ethnicity analysis using pre-computed reference model</h2>
<p>Analysis of target individuals genotype data using a reference model
built from 1,000 Genome Project genotype data. Genotype data for 10,000
exonic SNPs are provided in input to EthSEQ in VCF format while
reference model selected among the set of pre-computed reference models.
Reference model Gencode.Exome is used considering hg38 human reference
genome assembly and All populations (EUR, AFR, AMR, SAS, EAS). The
complete list of reference models can be visualized using the function
getModelsList().</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(EthSEQ)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="do">## View all available reference models</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="fu">getModelsList</span>()</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="do">## Run the analysis</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ethseq.Analysis</span>(</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">target.vcf =</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;Samples.HGDP.10000SNPs.vcf&quot;</span>,<span class="at">package=</span><span class="st">&quot;EthSEQ&quot;</span>),</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">model.available =</span> <span class="st">&quot;Gencode.Exome&quot;</span>,</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">model.assembly =</span> <span class="st">&quot;hg38&quot;</span>,</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">model.pop =</span> <span class="st">&quot;All&quot;</span>,</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">out.dir =</span> <span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;EthSEQ_Analysis/&quot;</span>),</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose=</span><span class="cn">TRUE</span>,</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">cores =</span><span class="dv">1</span>,</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">composite.model.call.rate =</span> <span class="dv">1</span>,</span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">space =</span> <span class="st">&quot;3D&quot;</span>)</span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a><span class="do">## Delete analysis folder</span></span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a><span class="fu">unlink</span>(<span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;EthSEQ_Analysis/&quot;</span>),<span class="at">recursive=</span><span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="perform-ethnicity-analysis-from-bam-files-list" class="section level2">
<h2>Perform ethnicity analysis from BAM files list</h2>
<p>Analysis of individual NA07357 from 1,000 Genome Project using a
reference model built from 1,000 Genome Project individual’s genotype
data. Genotype data for 10,000 SNPs included in Agilent Sure Select v2
captured regions are provided in input to EthSEQ with a BAM file.
reference model is provided in GDS format and describes genotype data
for 1,000 Genome Project individuls for the same SNPs set. Note than the
BAM given in input to EthSEQ is a toy BAM file containing only reads
overlapping the positions of the 10,000 SNPs considered in the
analysis.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(EthSEQ)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Download BAM file used in the analysis</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(<span class="st">&quot;https://github.com/cibiobcg/EthSEQ_Data/raw/master/BAM/HGDP00228.sub_GRCh38.bam&quot;</span>,</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;HGDP00228.sub_GRCh38.bam&quot;</span>))</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(<span class="st">&quot;https://github.com/cibiobcg/EthSEQ_Data/raw/master/BAM/HGDP00228.sub_GRCh38.bam.bai&quot;</span>,</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;HGDP00228.sub_GRCh38.bam.bai&quot;</span>))</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(<span class="st">&quot;https://github.com/cibiobcg/EthSEQ_Data/raw/master/BAM/HGDP01200.sub_GRCh38.bam&quot;</span>,</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;HGDP01200.sub_GRCh38.bam&quot;</span>))</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(<span class="st">&quot;https://github.com/cibiobcg/EthSEQ_Data/raw/master/BAM/HGDP01200.sub_GRCh38.bam.bai&quot;</span>,</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;HGDP01200.sub_GRCh38.bam.bai&quot;</span>))</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(<span class="st">&quot;https://github.com/cibiobcg/EthSEQ_Data/raw/master/BAM/HGDP01201.sub_GRCh38.bam&quot;</span>,</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;HGDP01201.sub_GRCh38.bam&quot;</span>))</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(<span class="st">&quot;https://github.com/cibiobcg/EthSEQ_Data/raw/master/BAM/HGDP01201.sub_GRCh38.bam.bai&quot;</span>,</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;HGDP01201.sub_GRCh38.bam.bai&quot;</span>))</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a><span class="do">## Create BAM files list </span></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(<span class="fu">c</span>(<span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;HGDP00228.sub_GRCh38.bam&quot;</span>),</span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>        <span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;HGDP01200.sub_GRCh38.bam&quot;</span>),</span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>        <span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;HGDP01201.sub_GRCh38.bam&quot;</span>)),</span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>        <span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;BAMs_List.txt&quot;</span>))</span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a><span class="do">## Run the analysis</span></span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a><span class="fu">ethseq.Analysis</span>(</span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">bam.list =</span> <span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;BAMs_List.txt&quot;</span>),</span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">model.available =</span> <span class="st">&quot;Gencode.Exome&quot;</span>,</span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">out.dir =</span> <span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;EthSEQ_Analysis/&quot;</span>),</span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">cores =</span> <span class="dv">1</span>,</span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">aseq.path =</span> <span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;EthSEQ_Analysis/&quot;</span>),</span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">run.genotype =</span> <span class="cn">TRUE</span>,</span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">mbq =</span> <span class="dv">20</span>,</span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">mrq =</span> <span class="dv">20</span>,</span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a>  <span class="at">mdc =</span> <span class="dv">10</span>,</span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a>  <span class="at">composite.model.call.rate =</span> <span class="dv">1</span>,</span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a>  <span class="at">space =</span> <span class="st">&quot;3D&quot;</span>,</span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a>  <span class="at">bam.chr.encoding =</span> <span class="cn">TRUE</span>) <span class="co"># chromosome names encoded without &quot;chr&quot; prefix in BAM files</span></span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-39"><a href="#cb9-39" aria-hidden="true" tabindex="-1"></a><span class="do">## Delete analysis folder</span></span>
<span id="cb9-40"><a href="#cb9-40" aria-hidden="true" tabindex="-1"></a><span class="fu">unlink</span>(<span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;EthSEQ_Analysis/&quot;</span>),<span class="at">recursive=</span><span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="perform-ethnicity-analysis-using-multi-step-refinement" class="section level2">
<h2>Perform ethnicity analysis using multi-step refinement</h2>
<p>Multi-step refinement analysis using a pre-computed reference model.
Genotype data for 10,000 exonic SNPs are provided in input to EthSEQ in
VCF format. Multi-step refinement tree is constructed as a matrix.
Non-empty cells in columns i contains parent nodes for non-empty cells
in columns i+1. Ancestry groups in child nodes should be included in
parent nodes, while siblings node ancestry groups should be disjoint.
Consult EthSEQ papers for more details.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(EthSEQ)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Create multi-step refinement matrix</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>m <span class="ot">=</span> <span class="fu">matrix</span>(<span class="st">&quot;&quot;</span>,<span class="at">ncol=</span><span class="dv">2</span>,<span class="at">nrow=</span><span class="dv">2</span>)</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>m[<span class="dv">1</span>,<span class="dv">1</span>] <span class="ot">=</span> <span class="st">&quot;EUR|AFR|AMR&quot;</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>m[<span class="dv">2</span>,<span class="dv">2</span>] <span class="ot">=</span> <span class="st">&quot;EUR|AMR&quot;</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="do">## Run the analysis on a toy example with only 10000 SNPs</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ethseq.Analysis</span>(</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">target.vcf =</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>,<span class="st">&quot;Samples.HGDP.10000SNPs.vcf&quot;</span>,<span class="at">package=</span><span class="st">&quot;EthSEQ&quot;</span>),</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">out.dir =</span> <span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;EthSEQ_Analysis/&quot;</span>),</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">model.available =</span> <span class="st">&quot;Gencode.Exome&quot;</span>,</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">refinement.analysis =</span> m,</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">composite.model.call.rate =</span> <span class="dv">1</span>,</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">space =</span> <span class="st">&quot;3D&quot;</span>)</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a><span class="do">## Delete analysis folder</span></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a><span class="fu">unlink</span>(<span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;EthSEQ_Analysis/&quot;</span>),<span class="at">recursive=</span><span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="create-a-reference-model-from-multiple-vcf-genotype-data-files" class="section level2">
<h2>Create a reference model from multiple VCF genotype data files</h2>
<p>Construction of a reference model from two genotype data files in VCF
format and a corresponding annotation files which described ancestry and
sex of each sample contained in the genotype data files.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(EthSEQ)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="do">### Load list of VCF files paths</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>vcf.files <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>,<span class="st">&quot;RefSample1.vcf&quot;</span>, <span class="at">package=</span><span class="st">&quot;EthSEQ&quot;</span>),</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>                <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>,<span class="st">&quot;RefSample2.vcf&quot;</span>, <span class="at">package=</span><span class="st">&quot;EthSEQ&quot;</span>))</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="do">### Load samples annotations</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>annot.samples <span class="ot">=</span> <span class="fu">read.delim</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>,<span class="st">&quot;Annotations_Test_v3.txt&quot;</span>,<span class="at">package=</span><span class="st">&quot;EthSEQ&quot;</span>))</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="do">### Create reference model</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a><span class="fu">ethseq.RM</span>(</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">vcf.fn =</span> vcf.files,</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotations =</span> annot.samples,</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">out.dir =</span> <span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;EthSEQ_Analysis/&quot;</span>),</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">model.name =</span> <span class="st">&quot;Reference.Model&quot;</span>,</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">bed.fn =</span> <span class="cn">NA</span>,</span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">call.rate =</span> <span class="dv">1</span>,</span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">cores =</span> <span class="dv">1</span>)</span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a><span class="do">## Delete example file</span></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a><span class="fu">unlink</span>(<span class="fu">file.path</span>(<span class="fu">tempdir</span>(),<span class="st">&quot;EthSEQ_Analysis/&quot;</span>),<span class="at">recursive=</span><span class="cn">TRUE</span>)</span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
