<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Alessandro Romanel" />

<meta name="date" content="2018-03-16" />

<title>EthSEQ: Ethnicity Annotation from Whole Exome Sequencing Data</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; }
code > span.dt { color: #902000; }
code > span.dv { color: #40a070; }
code > span.bn { color: #40a070; }
code > span.fl { color: #40a070; }
code > span.ch { color: #4070a0; }
code > span.st { color: #4070a0; }
code > span.co { color: #60a0b0; font-style: italic; }
code > span.ot { color: #007020; }
code > span.al { color: #ff0000; font-weight: bold; }
code > span.fu { color: #06287e; }
code > span.er { color: #ff0000; font-weight: bold; }
</style>



<link href="data:text/css,body%20%7B%0A%20%20background%2Dcolor%3A%20%23fff%3B%0A%20%20margin%3A%201em%20auto%3B%0A%20%20max%2Dwidth%3A%20700px%3B%0A%20%20overflow%3A%20visible%3B%0A%20%20padding%2Dleft%3A%202em%3B%0A%20%20padding%2Dright%3A%202em%3B%0A%20%20font%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0A%20%20font%2Dsize%3A%2014px%3B%0A%20%20line%2Dheight%3A%201%2E35%3B%0A%7D%0A%0A%23header%20%7B%0A%20%20text%2Dalign%3A%20center%3B%0A%7D%0A%0A%23TOC%20%7B%0A%20%20clear%3A%20both%3B%0A%20%20margin%3A%200%200%2010px%2010px%3B%0A%20%20padding%3A%204px%3B%0A%20%20width%3A%20400px%3B%0A%20%20border%3A%201px%20solid%20%23CCCCCC%3B%0A%20%20border%2Dradius%3A%205px%3B%0A%0A%20%20background%2Dcolor%3A%20%23f6f6f6%3B%0A%20%20font%2Dsize%3A%2013px%3B%0A%20%20line%2Dheight%3A%201%2E3%3B%0A%7D%0A%20%20%23TOC%20%2Etoctitle%20%7B%0A%20%20%20%20font%2Dweight%3A%20bold%3B%0A%20%20%20%20font%2Dsize%3A%2015px%3B%0A%20%20%20%20margin%2Dleft%3A%205px%3B%0A%20%20%7D%0A%0A%20%20%23TOC%20ul%20%7B%0A%20%20%20%20padding%2Dleft%3A%2040px%3B%0A%20%20%20%20margin%2Dleft%3A%20%2D1%2E5em%3B%0A%20%20%20%20margin%2Dtop%3A%205px%3B%0A%20%20%20%20margin%2Dbottom%3A%205px%3B%0A%20%20%7D%0A%20%20%23TOC%20ul%20ul%20%7B%0A%20%20%20%20margin%2Dleft%3A%20%2D2em%3B%0A%20%20%7D%0A%20%20%23TOC%20li%20%7B%0A%20%20%20%20line%2Dheight%3A%2016px%3B%0A%20%20%7D%0A%0Atable%20%7B%0A%20%20margin%3A%201em%20auto%3B%0A%20%20border%2Dwidth%3A%201px%3B%0A%20%20border%2Dcolor%3A%20%23DDDDDD%3B%0A%20%20border%2Dstyle%3A%20outset%3B%0A%20%20border%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0A%20%20border%2Dwidth%3A%202px%3B%0A%20%20padding%3A%205px%3B%0A%20%20border%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0A%20%20border%2Dwidth%3A%201px%3B%0A%20%20border%2Dstyle%3A%20inset%3B%0A%20%20line%2Dheight%3A%2018px%3B%0A%20%20padding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0A%20%20border%2Dleft%2Dstyle%3A%20none%3B%0A%20%20border%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0A%20%20background%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0A%0Ap%20%7B%0A%20%20margin%3A%200%2E5em%200%3B%0A%7D%0A%0Ablockquote%20%7B%0A%20%20background%2Dcolor%3A%20%23f6f6f6%3B%0A%20%20padding%3A%200%2E25em%200%2E75em%3B%0A%7D%0A%0Ahr%20%7B%0A%20%20border%2Dstyle%3A%20solid%3B%0A%20%20border%3A%20none%3B%0A%20%20border%2Dtop%3A%201px%20solid%20%23777%3B%0A%20%20margin%3A%2028px%200%3B%0A%7D%0A%0Adl%20%7B%0A%20%20margin%2Dleft%3A%200%3B%0A%7D%0A%20%20dl%20dd%20%7B%0A%20%20%20%20margin%2Dbottom%3A%2013px%3B%0A%20%20%20%20margin%2Dleft%3A%2013px%3B%0A%20%20%7D%0A%20%20dl%20dt%20%7B%0A%20%20%20%20font%2Dweight%3A%20bold%3B%0A%20%20%7D%0A%0Aul%20%7B%0A%20%20margin%2Dtop%3A%200%3B%0A%7D%0A%20%20ul%20li%20%7B%0A%20%20%20%20list%2Dstyle%3A%20circle%20outside%3B%0A%20%20%7D%0A%20%20ul%20ul%20%7B%0A%20%20%20%20margin%2Dbottom%3A%200%3B%0A%20%20%7D%0A%0Apre%2C%20code%20%7B%0A%20%20background%2Dcolor%3A%20%23f7f7f7%3B%0A%20%20border%2Dradius%3A%203px%3B%0A%20%20color%3A%20%23333%3B%0A%20%20white%2Dspace%3A%20pre%2Dwrap%3B%20%20%20%20%2F%2A%20Wrap%20long%20lines%20%2A%2F%0A%7D%0Apre%20%7B%0A%20%20border%2Dradius%3A%203px%3B%0A%20%20margin%3A%205px%200px%2010px%200px%3B%0A%20%20padding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0A%20%20background%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0A%0Acode%20%7B%0A%20%20font%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0A%20%20font%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0A%20%20padding%3A%202px%200px%3B%0A%7D%0A%0Adiv%2Efigure%20%7B%0A%20%20text%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0A%20%20background%2Dcolor%3A%20%23FFFFFF%3B%0A%20%20padding%3A%202px%3B%0A%20%20border%3A%201px%20solid%20%23DDDDDD%3B%0A%20%20border%2Dradius%3A%203px%3B%0A%20%20border%3A%201px%20solid%20%23CCCCCC%3B%0A%20%20margin%3A%200%205px%3B%0A%7D%0A%0Ah1%20%7B%0A%20%20margin%2Dtop%3A%200%3B%0A%20%20font%2Dsize%3A%2035px%3B%0A%20%20line%2Dheight%3A%2040px%3B%0A%7D%0A%0Ah2%20%7B%0A%20%20border%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0A%20%20padding%2Dtop%3A%2010px%3B%0A%20%20padding%2Dbottom%3A%202px%3B%0A%20%20font%2Dsize%3A%20145%25%3B%0A%7D%0A%0Ah3%20%7B%0A%20%20border%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0A%20%20padding%2Dtop%3A%2010px%3B%0A%20%20font%2Dsize%3A%20120%25%3B%0A%7D%0A%0Ah4%20%7B%0A%20%20border%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0A%20%20margin%2Dleft%3A%208px%3B%0A%20%20font%2Dsize%3A%20105%25%3B%0A%7D%0A%0Ah5%2C%20h6%20%7B%0A%20%20border%2Dbottom%3A%201px%20solid%20%23ccc%3B%0A%20%20font%2Dsize%3A%20105%25%3B%0A%7D%0A%0Aa%20%7B%0A%20%20color%3A%20%230033dd%3B%0A%20%20text%2Ddecoration%3A%20none%3B%0A%7D%0A%20%20a%3Ahover%20%7B%0A%20%20%20%20color%3A%20%236666ff%3B%20%7D%0A%20%20a%3Avisited%20%7B%0A%20%20%20%20color%3A%20%23800080%3B%20%7D%0A%20%20a%3Avisited%3Ahover%20%7B%0A%20%20%20%20color%3A%20%23BB00BB%3B%20%7D%0A%20%20a%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0A%20%20%20%20text%2Ddecoration%3A%20underline%3B%20%7D%0A%20%20a%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0A%20%20%20%20text%2Ddecoration%3A%20underline%3B%20%7D%0A%0A%2F%2A%20Class%20described%20in%20https%3A%2F%2Fbenjeffrey%2Ecom%2Fposts%2Fpandoc%2Dsyntax%2Dhighlighting%2Dcss%0A%20%20%20Colours%20from%20https%3A%2F%2Fgist%2Egithub%2Ecom%2Frobsimmons%2F1172277%20%2A%2F%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%2F%2A%20Keyword%20%2A%2F%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%2F%2A%20DataType%20%2A%2F%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%2F%2A%20DecVal%20%28decimal%20values%29%20%2A%2F%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20BaseN%20%2A%2F%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20Float%20%2A%2F%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20Char%20%2A%2F%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20String%20%2A%2F%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%2F%2A%20Comment%20%2A%2F%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%2F%2A%20OtherToken%20%2A%2F%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%2F%2A%20AlertToken%20%2A%2F%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%2F%2A%20Function%20calls%20%2A%2F%20%0Acode%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%2F%2A%20ErrorTok%20%2A%2F%0A%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">EthSEQ: Ethnicity Annotation from Whole Exome Sequencing Data</h1>
<h4 class="author"><em>Alessandro Romanel</em></h4>
<h4 class="date"><em>2018-03-16</em></h4>



<p>Whole exome sequencing (WES) is widely utilized both in translational cancer genomics studies and in the setting of precision medicine. Stratification of individual’s ethnicity is fundamental for the correct interpretation of personal genomic variation impact. We implemented EthSEQ to provide reliable and rapid ethnicity annotation from whole exome sequencing individual’s data. EthSEQ can be integrated into any WES based processing pipeline and exploits multi-core capabilities.</p>
<p>EthSEQ requires genotype data at SNPs positions for a set of individuals with known ethnicity (the reference model) and either a list of BAM files or genotype data (in VCF format) of individuals with unknown ethnicity. EthSEQ annotates the ethnicity of each individual using an automated procedure and returns detailed information about individual’s inferred ethnicity, including aggregated visual reports.</p>
<hr />
<div id="perform-ethnicity-analysis-with-individuals-genotype-data-from-vcf-file" class="section level2">
<h2>Perform ethnicity analysis with individuals genotype data from VCF file</h2>
<p>Analysis of 6 individuals from 1,000 Genome Project using a reference model built from 1,000 Genome Project individual’s genotype data. Genotype data for 10,000 SNPs included in Agilent Sure Select v2 captured regions are provided in input to EthSEQ in VCF format while reference model is provided in GDS format and describes genotype data for 1,000 Genome Project individuls for the same SNPs set.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(EthSEQ)

out.dir =<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">tempdir</span>(),<span class="st">&quot;EthSEQ_Analysis/&quot;</span>)

## Run the analysis
<span class="kw">ethseq.Analysis</span>(
  <span class="dt">target.vcf =</span> <span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;Samples_SS2_10000SNPs.vcf&quot;</span>,<span class="dt">package=</span><span class="st">&quot;EthSEQ&quot;</span>),
  <span class="dt">out.dir =</span> out.dir,
  <span class="dt">model.gds =</span> <span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;Reference_SS2_10000SNPs.gds&quot;</span>,<span class="dt">package=</span><span class="st">&quot;EthSEQ&quot;</span>),
  <span class="dt">verbose=</span><span class="ot">TRUE</span>,
  <span class="dt">composite.model.call.rate =</span> <span class="dv">1</span>,
  <span class="dt">space =</span> <span class="st">&quot;3D&quot;</span>) <span class="co"># Default space is 2D</span></code></pre>
<pre><code>## [2018-03-16 14:33:37] Running EthSEQ
## [2018-03-16 14:33:37] Working directory: /tmp/RtmpKCaudq/EthSEQ_Analysis/
## [2018-03-16 14:33:37] Create /tmp/RtmpKCaudq/EthSEQ_Analysis/ folder
## [2018-03-16 14:33:37] Create target model from VCF
## [2018-03-16 14:33:38] Create aggregated model
## [2018-03-16 14:33:39] Perform PCA on aggregated model
## Principal Component Analysis (PCA) on genotypes:
## Excluding 0 SNP on non-autosomes
## Excluding 53 SNPs (monomorphic: TRUE, MAF: NaN, missing rate: 0)
## Working space: 606 samples, 9,947 SNPs
##     using 1 (CPU) core
## PCA:    the sum of all selected genotypes (0,1,2) = 1530755
## CPU capabilities: Double-Precision SSE2
## Fri Mar 16 14:33:39 2018    (internal increment: 2100)
## 
[..................................................]  0%, ETC: ---    
[==================================================] 100%, completed in 0s
## Fri Mar 16 14:33:39 2018    Begin (eigenvalues and eigenvectors)
## Fri Mar 16 14:33:39 2018    Done.
## [2018-03-16 14:33:39] Infer ethnicities</code></pre>
<pre><code>## 
##      PLEASE NOTE:  As of version 0.3-5, no degenerate (zero area) 
##      regions are returned with the &quot;Qt&quot; option since the R 
##      code removes them from the triangulation. 
##      See help(&quot;delaunayn&quot;).</code></pre>
<pre><code>## [2018-03-16 14:33:40] Print annotations
## [2018-03-16 14:33:40] Plot visual report</code></pre>
<pre><code>## [2018-03-16 14:33:40] Computation end</code></pre>
<pre><code>## [1] TRUE</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Load and display computed ethnicity annotations</span>
ethseq.annotations =<span class="st"> </span><span class="kw">read.delim</span>(<span class="kw">file.path</span>(out.dir,<span class="st">&quot;Report.txt&quot;</span>),<span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>,<span class="dt">as.is=</span><span class="ot">TRUE</span>,<span class="dt">header=</span><span class="ot">TRUE</span>)
<span class="kw">head</span>(ethseq.annotations)</code></pre>
<pre><code>##   sample.id pop    type            contribution
## 1   NA18972 EAS  INSIDE                        
## 2   NA19247 AFR  INSIDE                        
## 3   HG02922 AFR CLOSEST AFR(85.11%)|SAS(14.89%)
## 4   NA19248 AFR  INSIDE                        
## 5   NA18951 EAS CLOSEST EAS(81.93%)|EUR(18.07%)
## 6   NA21127 SAS  INSIDE</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">## Delete analysis folder
<span class="kw">unlink</span>(out.dir,<span class="dt">recursive=</span><span class="ot">TRUE</span>)</code></pre>
</div>
<div id="perform-ethnicity-analysis-using-pre-computed-reference-model" class="section level2">
<h2>Perform ethnicity analysis using pre-computed reference model</h2>
<p>Analysis of 6 individuals from 1,000 Genome Project using a reference model built from 1,000 Genome Project individual’s genotype data. Genotype data for 123,292 SNPs included in Agilent Sure Select v2 captured regions are provided in input to EthSEQ in VCF format while reference model selected among the set of pre-computed reference model. Reference model SS2.Major refers to the reference model built from 1550 individuals (from AFR, EUR, SAS and EAS major populations) from 1,000 Genome Project and considering 123,292 SNPs included in Agilent Sure Select v2 captured regions. Note that a reference model version called SS2 considering gentoype data for more than 2,000 individuals from 1,000 Genome Project is also available.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(EthSEQ)

data.dir =<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">tempdir</span>(),<span class="st">&quot;EthSEQ_Data/&quot;</span>)
out.dir =<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">tempdir</span>(),<span class="st">&quot;EthSEQ_Analysis/&quot;</span>)

## Download genotype data in VCF format
<span class="kw">dir.create</span>(data.dir)
<span class="kw">download.file</span>(<span class="st">&quot;https://github.com/aromanel/EthSEQ_Data/raw/master/Sample_SS2.vcf&quot;</span>,
  <span class="dt">destfile =</span> <span class="kw">file.path</span>(data.dir,<span class="st">&quot;Sample_SS2.vcf&quot;</span>))

## Run the analysis
<span class="kw">ethseq.Analysis</span>(
  <span class="dt">target.vcf =</span>  <span class="kw">file.path</span>(data.dir,<span class="st">&quot;Sample_SS2.vcf&quot;</span>),
  <span class="dt">out.dir =</span> out.dir,
  <span class="dt">model.available =</span> <span class="st">&quot;SS2.Major&quot;</span>,
  <span class="dt">model.folder =</span> data.dir,
  <span class="dt">verbose=</span><span class="ot">TRUE</span>,
  <span class="dt">composite.model.call.rate =</span> <span class="dv">1</span>,
  <span class="dt">space =</span> <span class="st">&quot;3D&quot;</span>) <span class="co"># Default space is 2D</span>

## Delete analysis folder
<span class="kw">unlink</span>(data.dir,<span class="dt">recursive=</span><span class="ot">TRUE</span>)
<span class="kw">unlink</span>(out.dir,<span class="dt">recursive=</span><span class="ot">TRUE</span>)</code></pre>
</div>
<div id="perform-ethnicity-analysis-from-bam-files-list" class="section level2">
<h2>Perform ethnicity analysis from BAM files list</h2>
<p>Analysis of individual NA07357 from 1,000 Genome Project using a reference model built from 1,000 Genome Project individual’s genotype data. Genotype data for 10,000 SNPs included in Agilent Sure Select v2 captured regions are provided in input to EthSEQ with a BAM file. reference model is provided in GDS format and describes genotype data for 1,000 Genome Project individuls for the same SNPs set. Note than the BAM given in input to EthSEQ is a toy BAM file containing only reads overlapping the positions of the 10,000 SNPs considered in the analysis.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(EthSEQ)

data.dir =<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">tempdir</span>(),<span class="st">&quot;EthSEQ_Data&quot;</span>)
out.dir =<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">tempdir</span>(),<span class="st">&quot;EthSEQ_Analysis&quot;</span>)

## Download BAM file used in the analysis
<span class="kw">dir.create</span>(data.dir)
<span class="kw">download.file</span>(
 <span class="st">&quot;https://github.com/aromanel/EthSEQ_Data/raw/master/NA07357_only10000SNPs.bam&quot;</span>,
 <span class="dt">destfile =</span> <span class="kw">file.path</span>(data.dir,<span class="st">&quot;Sample.bam&quot;</span>))
<span class="kw">download.file</span>(
 <span class="st">&quot;https://github.com/aromanel/EthSEQ_Data/raw/master/NA07357_only10000SNPs.bam.bai&quot;</span>,
 <span class="dt">destfile =</span> <span class="kw">file.path</span>(data.dir,<span class="st">&quot;Sample.bam.bai&quot;</span>))

## Create BAM files list 
<span class="kw">write</span>(<span class="kw">file.path</span>(data.dir,<span class="st">&quot;Sample.bam&quot;</span>),<span class="kw">file.path</span>(data.dir,<span class="st">&quot;BAMs_List.txt&quot;</span>))

## Run the analysis
<span class="kw">ethseq.Analysis</span>(
  <span class="dt">bam.list =</span> <span class="kw">file.path</span>(data.dir,<span class="st">&quot;BAMs_List.txt&quot;</span>),
  <span class="dt">out.dir =</span> out.dir,
  <span class="dt">model.gds =</span> <span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>,<span class="st">&quot;Reference_SS2_10000SNPs.gds&quot;</span>,
     <span class="dt">package=</span><span class="st">&quot;EthSEQ&quot;</span>),
  <span class="dt">verbose=</span><span class="ot">TRUE</span>,
  <span class="dt">aseq.path =</span> out.dir,
  <span class="dt">mbq=</span><span class="dv">20</span>,
  <span class="dt">mrq=</span><span class="dv">20</span>,
  <span class="dt">mdc=</span><span class="dv">10</span>,
  <span class="dt">run.genotype =</span> <span class="ot">TRUE</span>,
  <span class="dt">composite.model.call.rate =</span> <span class="dv">1</span>,
  <span class="dt">cores=</span><span class="dv">1</span>,
  <span class="dt">bam.chr.encoding =</span> <span class="ot">FALSE</span>) <span class="co"># chromosome names encoded without &quot;chr&quot; prefix in BAM files</span>

## Delete analysis folder
<span class="kw">unlink</span>(data.dir,<span class="dt">recursive=</span><span class="ot">TRUE</span>)
<span class="kw">unlink</span>(out.dir,<span class="dt">recursive=</span><span class="ot">TRUE</span>)</code></pre>
</div>
<div id="perform-ethnicity-analysis-using-multi-step-refinement" class="section level2">
<h2>Perform ethnicity analysis using multi-step refinement</h2>
<p>Multi-step refinement Analysis of individuals from 1,000 Genome Project using a reference model built from 1,000 Genome Project individual’s genotype data (set of analysed individuals and individuals used for the reference model are disjoint). Genotype data for 10,000 SNPs included in Agilent Sure Select v2 captured regions are provided in input to EthSEQ in GDS format while reference model is provided in GDS format and describes genotype data for 1,000 Genome Project reference individuls for the same SNPs set. Multi-step refinement tree is constructed as matrix. Non-empty cells in columns i contains parent nodes for non-empty cells in columns i+1. Ethnic groups in child nodes should be included in parent nodes, while siblings node ethnic groups should be disjoint. Consult EthSEQ paper supplementary material for more complicated examples.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(EthSEQ)

out.dir =<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">tempdir</span>(),<span class="st">&quot;EthSEQ_Analysis&quot;</span>)
data.dir =<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">tempdir</span>(),<span class="st">&quot;EthSEQ_Data&quot;</span>)

## Download genotype data in VCF format
<span class="kw">dir.create</span>(data.dir)
<span class="kw">download.file</span>(<span class="st">&quot;https://github.com/aromanel/EthSEQ_Data/raw/master/Target_SS2_10000SNPs.gds&quot;</span>,
  <span class="dt">destfile =</span> <span class="kw">file.path</span>(data.dir,<span class="st">&quot;Target_SS2_10000SNPs.gds&quot;</span>))

## Create multi-step refinement matrix
m =<span class="st"> </span><span class="kw">matrix</span>(<span class="st">&quot;&quot;</span>,<span class="dt">ncol=</span><span class="dv">2</span>,<span class="dt">nrow=</span><span class="dv">2</span>)
m[<span class="dv">1</span>,<span class="dv">1</span>] =<span class="st"> &quot;SAS|EUR|EAS&quot;</span>
m[<span class="dv">2</span>,<span class="dv">2</span>] =<span class="st"> &quot;SAS|EUR&quot;</span>

## Run the analysis on a toy example with only 10000 SNPs
<span class="kw">ethseq.Analysis</span>(
  <span class="dt">target.gds =</span> <span class="kw">file.path</span>(data.dir,<span class="st">&quot;Target_SS2_10000SNPs.gds&quot;</span>),
  <span class="dt">out.dir =</span> out.dir,
  <span class="dt">model.gds =</span> <span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>,<span class="st">&quot;Reference_SS2_10000SNPs.gds&quot;</span>,
    <span class="dt">package=</span><span class="st">&quot;EthSEQ&quot;</span>),
  <span class="dt">verbose=</span><span class="ot">TRUE</span>,
  <span class="dt">composite.model.call.rate =</span> <span class="dv">1</span>,
  <span class="dt">refinement.analysis =</span> m,
  <span class="dt">space=</span><span class="st">&quot;3D&quot;</span>)

## Delete analysis folder
<span class="kw">unlink</span>(out.dir,<span class="dt">recursive=</span><span class="ot">TRUE</span>)</code></pre>
</div>
<div id="create-a-reference-model-from-multiple-vcf-genotype-data-files" class="section level2">
<h2>Create a reference model from multiple VCF genotype data files</h2>
<p>Construction of a reference model from two genotype data files in VCF format and a corresponding annotation files which described ethnicity and sex of each sample contained in the genotype data files.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(EthSEQ)

out.dir =<span class="st"> </span><span class="kw">tempdir</span>()
<span class="kw">dir.create</span>(out.dir)

### Load list of VCF files paths
vcf.files =<span class="st"> </span>
<span class="st">  </span><span class="kw">c</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;VCF_Test_1.vcf&quot;</span>, <span class="dt">package=</span><span class="st">&quot;EthSEQ&quot;</span>),
    <span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;VCF_Test_2.vcf&quot;</span>, <span class="dt">package=</span><span class="st">&quot;EthSEQ&quot;</span>))

### Load samples annotations
annot.samples =<span class="st"> </span><span class="kw">read.delim</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;Annotations_Test.txt&quot;</span>,
    <span class="dt">package=</span><span class="st">&quot;EthSEQ&quot;</span>))

### Create reference model
<span class="kw">ethseq.RM</span>(
  <span class="dt">vcf.fn =</span> vcf.files,
  <span class="dt">annotations =</span> annot.samples,
  <span class="dt">out.dir =</span> out.dir,
  <span class="dt">model.name =</span> <span class="st">&quot;Reference.Model&quot;</span>,
  <span class="dt">bed.fn =</span> <span class="ot">NA</span>,
  <span class="dt">call.rate =</span> <span class="dv">1</span>,
  <span class="dt">cores =</span> <span class="dv">1</span>)

## Delete example file
<span class="kw">unlink</span>(out.dir,<span class="dt">recursive=</span><span class="ot">TRUE</span>)</code></pre>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
